internal/provider/resource_access_model.go

```diff
@@ -221,52 +335,33 @@ func getAddedUsers(oldUsers, newUsers []string) []string {
 	return added
 }
 
-// resourceAccessModelDelete deletes the access model resource
-// Juju refers to deletions as "destroy" so we call the Destroy function of our client here rather than delete
-// This function remains named Delete for parity across the provider and to stick within terraform naming conventions
-func resourceAccessModelDelete(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
-	client := meta.(*juju.Client)
-
-	var diags diag.Diagnostics
+func (a *accessModelResource) ImportState(ctx context.Context, req resource.ImportStateRequest, resp *resource.ImportStateResponse) {
+	resource.ImportStatePassthroughID(ctx, path.Root("id"), req, resp)
+}
 
-	usersInterface := d.Get("users").([]interface{})
-	users := make([]string, len(usersInterface))
-	for i, v := range usersInterface {
-		users[i] = v.(string)
-	}
-	access := d.Get("access").(string)
+func addClientNotConfiguredError(diag *diag.Diagnostics, method string) {
+	diag.AddError(
+		"Provider Error, Client Not Configured",
+		fmt.Sprintf("Unable to %v access model resource. Expected configured Juju Client. "+
+			"Please report this issue to the provider developers.", method),
+	)
+}
 
-	err := client.Models.DestroyAccessModel(juju.DestroyAccessModelInput{
-		Model:  d.Id(),
-		Revoke: users,
-		Access: access,
-	})
-	if err != nil {
-		return diag.FromErr(err)
-	}
+func newAccessModelIDFrom(modelNameStr string, accessStr string, users []string) string {
+	return fmt.Sprintf("%s:%s:%s", modelNameStr, accessStr, strings.Join(users, ","))
+}
 
-	d.SetId("")
+func retrieveAccessModelDataFromID(model *accessModelResourceModel) (string, string, []string) {
+	resID := strings.Split(model.ID.ValueString(), ":")
+	stateUsers := strings.Split(resID[2], ",")
```

Comment:
Causing a panic when upgrading an existing deployment.  Check the length and error if not as expected.